entropylab/tests/test_retry.py

```diff
@@ -0,0 +1,42 @@
+from entropylab import PyNode, Graph
+from entropylab.api.graph import RetryBehavior
+from entropylab.logger import logger
+
+counter = 1
+
+
+def a():
+    global counter
+    if counter < 3:
+        counter = counter + 1
+        raise Exception("still not working")
+    counter = 1
+    return {"x": 1}
+
+
+def test_retry_behavior(caplog):
+    a1 = PyNode(
+        "a1",
+        a,
+        output_vars={"x"},
+        retry_on_error=RetryBehavior(backoff=2, wait_time=0.2),

```

Comment:
I would expect the number of max. num of attempts to be set via a parameter to this class but it seems it's set with the node "manifest" - i.e. the function `a()`. I think the counter behaviour is supposed to be part of RetryBehavior

Reply:
every parameter of retry behavior is exposed to the user, it just has default values if you don't specify:
```
@dataclass
class RetryBehavior:
    """
    Attributes:
        number_of_attempts: the maximum number of retries.
        wait_time: the initial wait time between each attempt [seconds].
        backoff: a factor that multiplies the delay on each attempt (1 is no backoff).
        added_delay: delay that is added on each attempt [seconds].
        max_wait_time: optional maximum delay.
    """

    number_of_attempts: float = 5
    wait_time: float = 10
    backoff: float = 2
    added_delay: float = 0
    max_wait_time: Optional[float] = None
```


in the node i just wanted it to fail some time to check that the retry really happens